samcli/local/common/runtime_template.py

```diff
@@ -162,3 +149,172 @@ def get_local_lambda_images_location(mapping, runtime):
     "python3.8": "Python36",
     "python3.9": "Python36",
 }
+
+CUSTOM_RUNTIME = ["provided.al2", "provided"]
+
+
+def is_custom_runtime(runtime):
+    """
+    validated if a runtime is custom or not
+
+    Parameters
+    ----------
+    runtime : str
+        runtime to be
+
+    Returns
+    -------
+    _type_
+        _description_
+    """
+    if not runtime:
+        return False
+    validation_result = get_custom_runtime_base_runtime(runtime)
+    return runtime in CUSTOM_RUNTIME or bool(validation_result in CUSTOM_RUNTIME)
+
+
+def get_custom_runtime_base_runtime(runtime):
+    base_runtime_list = re.findall("(?<=\\().*(?=\\))", runtime)
+    return base_runtime_list[0] if base_runtime_list else None
+
+
+def sort_runtimes(runtime_list):
+    """
+    Sort runtime in a descending order of the runtime name and ascending order of the runtime version
+
+    Parameters
+    ----------
+    runtime_list : list
+        list of runtime
+
+    Returns
+    -------
+    list
+        list of sorted runtime
+    """
+    runtime_list = verify_runtime_list_provided(runtime_list)
+    _sort_runtimes(runtime_list, 0, len(runtime_list) - 1)
```

Comment:
Any reason to not use the native `sort()` method but implement a sorting algorithm?

Reply:
the complexity of how we want runtimes to be displayed sorting. I am not sure the flexibility sort(). implementing mine allowed me more flexibility.

Reply:
A custom comparison can be passed to sorted as parameter called `key`, detail can be found here:(https://docs.python.org/3/howto/sorting.html#key-functions). If it work in this case, I suggest we just need to have a customized key function rather than write the sort function by ourselves.